infrastructure/src/main/java/org/corfudb/infrastructure/log/LogMetadata.java

```diff
@@ -144,4 +135,21 @@ public void prefixTrim(long address) {
             streamAddressMap.getValue().trim(address);
         }
     }
+
+    /**
+     * Update stream trim marks with the values computed after the complete log has been scanned.
+     */
+    public void updateStreamTrimMarks() {
```

Comment:
I think we can do this in-place,  i.e. put it in `if (initialize)` block and remove this method.

Reply:
Done.